sky/provision/kubernetes/utils.py

```diff
@@ -3179,7 +3184,30 @@ def filter_pods(namespace: str,
     pods = [
         pod for pod in pod_list.items if pod.metadata.deletion_timestamp is None
     ]
-    return {pod.metadata.name: pod for pod in pods}
+
+    # Sort pods by name, with workers sorted by their numeric suffix.
+    # This ensures consistent ordering (e.g., cluster-head, cluster-worker1,
+    # worker2, worker3, ...) even when Kubernetes API returns them in
+    # arbitrary order. This works even if there were somehow pod names other
+    # than head/worker ones, but that may be overkill.
+    def get_pod_sort_key(
+        pod: V1Pod
+    ) -> Union[Tuple[Literal[0], str], Tuple[Literal[1], int], Tuple[Literal[2],
+                                                                     str]]:
+        name = pod.metadata.name
+        if '-worker' in name:
+            try:
+                return (1, int(name.split('-worker')[-1]))
+            except (ValueError, IndexError):
+                return (2, name)
+        elif '-head' in name:
+            return (0, name)
+        else:
+            return (2, name)
```

Comment:
actually, this may fail to keep head node as the first if we have a cluster name `my-workers`?